src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -112,6 +113,19 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         LOG_INFO("Accuracy check is enabled.");
     }
 
+    // Sort out weights sharing routine
+    const std::string weights_bank_opt = m_cfg.get<::intel_npu::NPUW_WEIGHTS_BANK>();
+    if (!weights_bank_opt.empty()) {
+        if (m_cfg.get<::intel_npu::NPUW_FOLD>()) {
+            OPENVINO_THROW("NPUW can't utilize NPUW_WEIGHTS_BANK property with NPUW_FOLD enabled!");
+        }
+        if ("YES" != m_cfg.getString<::intel_npu::NPUW_CWAI>()) {
+            OPENVINO_THROW("NPUW can't utilize NPUW_WEIGHTS_BANK property without NPUW_CWAI enabled!");
+        }
```

Comment:
I don't think these inconsistencies are fatal to throw exceptions. Just raise WARNINGs where for combinations where it can't be used.